src/machines/modelingMachine.ts

```diff
@@ -1994,42 +1999,166 @@ export const modelingMachine = setup({
 
         // Extract inputs
         const ast = kclManager.ast
-        const { selection, thickness } = input
+        const { selection, thickness, nodeToEdit } = input
+        let variableName: string | undefined = undefined
+        let insertIndex: number | undefined = undefined
 
-        // Insert the thickness variable if it exists
-        if (
-          'variableName' in thickness &&
-          thickness.variableName &&
-          thickness.insertIndex !== undefined
-        ) {
-          const newBody = [...ast.body]
-          newBody.splice(
-            thickness.insertIndex,
-            0,
-            thickness.variableDeclarationAst
+        // If this is an edit flow, first we're going to remove the old extrusion
+        if (nodeToEdit && typeof nodeToEdit[1][0] === 'number') {
+          // Extract the plane name from the node to edit
+          const variableNode = getNodeFromPath<VariableDeclaration>(
+            ast,
+            nodeToEdit,
+            'VariableDeclaration'
           )
+          if (err(variableNode)) {
+            console.error('Error extracting name')
+          } else {
+            variableName = variableNode.node.declaration.id.name
+          }
+
+          // Removing the old statement
+          const newBody = [...ast.body]
+          newBody.splice(nodeToEdit[1][0], 1)
           ast.body = newBody
+          insertIndex = nodeToEdit[1][0]
+        }
+
+        // Turn the selection into the faces list
+        const clonedAstForGetExtrude = structuredClone(ast)
+        const faces: Expr[] = []
+        let pathToExtrudeNode: PathToNode | undefined = undefined
+        for (const graphSelection of selection.graphSelections) {
+          const extrudeLookupResult = getPathToExtrudeForSegmentSelection(
+            clonedAstForGetExtrude,
+            graphSelection,
+            engineCommandManager.artifactGraph
+          )
+          // TODO: this one returns empty paths on edit after creation in the cap e2e tests
+          // and *sometimes* when trying in the app manually. No idea why.
+          console.log(
+            'extrudeLookupResult',
+            JSON.stringify(extrudeLookupResult)
+          )
+          if (err(extrudeLookupResult)) {
+            const error = Error(
+              "Couldn't find extrude paths from getPathToExtrudeForSegmentSelection"
+            )
+            console.error(error)
```

Comment:
Why not the following? Doesn't the caller, at some point, log the error that we return, so why would we log it when we create it? Classic log-and-throw anti-pattern that duplicates errors.

```suggestion
            const error = new Error(
              "Couldn't find extrude paths from getPathToExtrudeForSegmentSelection",
              { cause: extrudeLookupResult}
            )
```

Reply:
@jtran thanks! I think I was just pissed I wasn't seeing them logged when I was testing. Will update

Reply:
Made the change